### PR TITLE
Fix typos and cleanup in various files

### PR DESCRIPTION
**Pull request by rex4539**

### CODE REVIEW

**Review by randolf — CHANGES_REQUESTED** [review]
These are excellent edits, but I need you to do the following...

Move all the changes in ``src/qt/locale/bitcoin_el_GR.ts`` to a separate Pull Request because this is in a different language than all the other changes (I don't know enough German to review German-language corrections).

Revert change in ``test/functional/p2p_fingerprint.py`` back to its original "its" (line 7).

Thank you.

UPDATE:  Thank you for making these changes.

NOTE:  The Greek (not German) Translation corrections can be found in PR https://github.com/bitcoin/bitcoin/pull/12598.

**Comment by rex4539** [comment]
The translation corrections are actually Greek :)

**Comment by rex4539** [comment]
I have no idea how idea how to move all the changes in src/qt/locale/bitcoin_el_GR.ts to a separate Pull Request (and then I guess remove that commit from the PR).

The commits are already squashed and googling didn't help.

If you can post the exact commands I need to paste into the command line, I would be happy to apply them.

Another solution would be to have the person who usually reviews the Greek translations review and approve.

**Comment by randolf** [comment]
@rex4539 Ah, Greek.  Ha!  Thanks for correcting me on that.

The problem is that I can't approve something that I don't understand.  I'll ask someone to take a look at this and give you some suggestions on the git commands to use.  In the meantime, there's also the #github channel in irc.freenode.net which has a lot very helpful people as well (they've been very helpful to me in the past).

**Comment by dongcarl** [comment]
@rex4539 on branch rex4539-typos:
```
git reset --soft 90a0aed51194ad82da8e011b96f9561c685e40b7
git reset HEAD src/qt/locale/bitcoin_el_GR.ts
<make other changes if needed>
git commit -m 'blah blah'
<then force push to your own remote>
```
It'll say that the `.ts` file is not staged for commit but that's cool.

**Review by randolf — APPROVED** [review]

File: src/core_write.cpp [hunk 1]
```diff
@@ -106,7 +106,7 @@ std::string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDeco
                         const unsigned char chSigHashType = vch.back();
                         if (mapSigHashTypes.count(chSigHashType)) {
                             strSigHashDecode = "[" + mapSigHashTypes.find(chSigHashType)->second + "]";
-                            vch.pop_back(); // remove the sighash type byte. it will be replaced by the decode.
+                            vch.pop_back(); // Remove the sighash type byte. It will be replaced by the decode.
```
Comment (practicalswift) [inline src/core_write.cpp]: Capitalization changes are unlikely to get consensus ACK, so please skip the changes made to this file.

File: src/secp256k1/src/java/org/bitcoin/NativeSecp256k1Test.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public static void testSecKeyVerifyPos() throws AssertFailException{
     }
 
     /**
-      * This tests secret key verify() for a invalid secretkey
+      * This tests secret key verify() for an invalid secretkey
```
Comment (practicalswift) [inline src/secp256k1/src/java/org/bitcoin/NativeSecp256k1Test.java]: These changes should be reported upstream. Please skip the changes made to this file.

File: src/univalue/lib/univalue_get.cpp [hunk 3]
```diff
@@ -35,7 +35,7 @@ bool ParseInt32(const std::string& str, int32_t *out)
     errno = 0; // strtol will not set errno if valid
     long int n = strtol(str.c_str(), &endp, 10);
     if(out) *out = (int32_t)n;
-    // Note that strtol returns a *long int*, so even if strtol doesn't report a over/underflow
+    // Note that strtol returns a *long int*, so even if strtol doesn't report an over/underflow
```
Comment (practicalswift) [inline src/univalue/lib/univalue_get.cpp]: These changes should be reported upstream. Please skip the changes made to this file.

File: test/functional/wallet_basic.py [hunk 4]
```diff
@@ -355,7 +355,7 @@ def run_test(self):
         for mode in [True, False]:
             self.nodes[0].ensure_ascii = mode
             # unicode check: Basic Multilingual Plane, Supplementary Plane respectively
-            for s in [u'рыба', u'𝅘𝅥𝅯']:
+            for s in [u'рыба', u'𝅘𝅥𝅯']:
```
Comment (practicalswift) [inline test/functional/wallet_basic.py]: Please revert this change. It looks accidental.

File: src/net_processing.cpp [hunk 5]
```diff
@@ -339,7 +339,7 @@ bool MarkBlockAsInFlight(NodeId nodeid, const uint256& hash, const CBlockIndex*
     CNodeState *state = State(nodeid);
     assert(state != nullptr);
 
-    // Short-circuit most stuff in case its from the same node
+    // Short-circuit most stuff in case it's from the same node
```
Comment (practicalswift) [inline src/net_processing.cpp]: Nit: Prefer "it is" to "it's". Applies to "its"/"it's" changes below.

File: src/validation.cpp [hunk 6]
```diff
@@ -949,7 +949,7 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
 
         // This transaction should only count for fee estimation if:
         // - it isn't a BIP 125 replacement transaction (may not be widely supported)
-        // - it's not being readded during a reorg which bypasses typical mempool fee limits
+        // - it's not being read during a reorg which bypasses typical mempool fee limits
```
Comment (practicalswift) [inline src/validation.cpp]: Shouldn't this be "re-added"?

File: src/wallet/rpcwallet.cpp [hunk 7]
```diff
@@ -1987,7 +1987,7 @@ UniValue listsinceblock(const JSONRPCRequest& request)
             "  ],\n"
             "  \"removed\": [\n"
             "    <structure is the same as \"transactions\" above, only present if include_removed=true>\n"
-            "    Note: transactions that were readded in the active chain will appear as-is in this array, and may thus have a positive confirmation count.\n"
+            "    Note: transactions that were read in the active chain will appear as-is in this array, and may thus have a positive confirmation count.\n"
```
Comment (practicalswift) [inline src/wallet/rpcwallet.cpp]: Shouldn't this be "re-added"?

File: src/test/util_tests.cpp [hunk 8]
```diff
@@ -754,7 +754,7 @@ BOOST_AUTO_TEST_CASE(test_LockDirectory)
     thr.join();
     BOOST_CHECK_EQUAL(threadresult, true);
 #ifndef WIN32
-    // Try to aquire lock in child process while we're holding it, this should fail.
+    // Try to acquire lock in child process while we're holding it, this should fail.
```
Comment (practicalswift) [inline src/test/util_tests.cpp]: Already reported in #12543.

File: src/test/util_tests.cpp [hunk 9]
```diff
@@ -765,7 +765,7 @@ BOOST_AUTO_TEST_CASE(test_LockDirectory)
     // Probing lock from our side now should succeed, but not hold on to the lock.
     BOOST_CHECK_EQUAL(LockDirectory(dirname, lockname, true), true);
 
-    // Try to acquire the lock in the child process, this should be succesful.
+    // Try to acquire the lock in the child process, this should be successful.
```
Comment (practicalswift) [inline src/test/util_tests.cpp]: Already reported in #12543.

File: test/functional/wallet_dump.py [hunk 10]
```diff
@@ -49,7 +49,7 @@ def read_dump(file_name, addrs, script_addrs, hd_master_addr_old):
                     # count key types
                     for addrObj in addrs:
                         if addrObj['address'] == addr.split(",")[0] and addrObj['hdkeypath'] == keypath and keytype == "label=":
-                            # a labled entry in the wallet should contain both a native address
+                            # a labeled entry in the wallet should contain both a native address
```
Comment (practicalswift) [inline test/functional/wallet_dump.py]: Already reported in #12543.

File: src/core_write.cpp (hunk 1 above)
Comment (randolf) [reply src/core_write.cpp]: While this change may appear to be purely cosmetic, I do think that it's okay since it is being included with a number of other changes and because the overall intention is to ensure greater consistency.

File: src/net_processing.cpp (hunk 5 above)
Comment (randolf) [reply src/net_processing.cpp]: Either "it is" or "it's" is acceptable to me.  Since you prefer "it is" I support this.

Comment (practicalswift) [reply src/net_processing.cpp]: I prefer "it is" since it is unambiguous. "It's" can be "it is" or "it has".

File: src/test/cuckoocache_tests.cpp [hunk 11]
```diff
@@ -163,7 +163,7 @@ void test_cache_erase(size_t megabytes)
     for (uint32_t i = (n_insert / 2); i < n_insert; ++i)
         set.insert(hashes_insert_copy[i]);
 
-    /** elements that we marked erased but that are still there */
+    /** elements that we marked erased but are still there */
```
Comment (practicalswift) [inline src/test/cuckoocache_tests.cpp]: Nit: Also change "marked erased" to "marked as erased".

File: src/wallet/wallet.cpp [hunk 12]
```diff
@@ -4184,8 +4184,8 @@ bool CWalletTx::AcceptToMemoryPool(const CAmount& nAbsurdFee, CValidationState&
     // We must set fInMempool here - while it will be re-set to true by the
     // entered-mempool callback, if we did not there would be a race where a
     // user could call sendmoney in a loop and hit spurious out of funds errors
-    // because we think that the transaction they just generated's change is
-    // unavailable as we're not yet aware its in mempool.
+    // because we think that the just generated transaction's change is
+    // unavailable as we're not yet aware it's in mempool.
```
Comment (practicalswift) [inline src/wallet/wallet.cpp]: A bit unclear. Perhaps change it to "as we're not yet aware that it is in the mempool"?

File: src/validation.cpp (hunk 6 above)
Comment (rex4539) [reply src/validation.cpp]: Fixed.

File: src/net_processing.cpp (hunk 5 above)
Comment (rex4539) [reply src/net_processing.cpp]: Fixed.

File: test/functional/wallet_basic.py (hunk 4 above)
Comment (rex4539) [reply test/functional/wallet_basic.py]: Hopefully fixed.

File: src/univalue/lib/univalue_get.cpp (hunk 3 above)
Comment (rex4539) [reply src/univalue/lib/univalue_get.cpp]: Fixed.

File: src/secp256k1/src/java/org/bitcoin/NativeSecp256k1Test.java (hunk 2 above)
Comment (rex4539) [reply src/secp256k1/src/java/org/bitcoin/NativeSecp256k1Test.java]: Fixed.

File: src/core_write.cpp (hunk 1 above)
Comment (rex4539) [reply src/core_write.cpp]: I didn't change this one. Let me know if you still need this changed.

File: src/wallet/rpcwallet.cpp (hunk 7 above)
Comment (rex4539) [reply src/wallet/rpcwallet.cpp]: Fixed.

File: src/test/util_tests.cpp (hunk 8 above)
Comment (rex4539) [reply src/test/util_tests.cpp]: So should I revert this one or just do nothing?

File: src/test/util_tests.cpp (hunk 9 above)
Comment (rex4539) [reply src/test/util_tests.cpp]: So should I revert this one or just do nothing?

File: test/functional/wallet_dump.py (hunk 10 above)
Comment (rex4539) [reply test/functional/wallet_dump.py]: So should I revert this one or just do nothing?

File: src/test/cuckoocache_tests.cpp (hunk 11 above)
Comment (rex4539) [reply src/test/cuckoocache_tests.cpp]: Fixed.

File: src/wallet/wallet.cpp (hunk 12 above)
Comment (rex4539) [reply src/wallet/wallet.cpp]: Fixed.

File: src/test/util_tests.cpp (hunk 9 above)
Comment (practicalswift) [reply src/test/util_tests.cpp]: Revert or rebase on top of `master` now that #12543 is merged.

File: test/functional/wallet_dump.py (hunk 10 above)
Comment (practicalswift) [reply test/functional/wallet_dump.py]: Revert or rebase on top of `master` now that #12543 is merged.

File: src/test/util_tests.cpp (hunk 8 above)
Comment (practicalswift) [reply src/test/util_tests.cpp]: Revert or rebase on top of master now that #12543 is merged.

File: test/functional/wallet_basic.py (hunk 4 above)
Comment (practicalswift) [reply test/functional/wallet_basic.py]: No, the change is still there. You can check it on https://github.com/bitcoin/bitcoin/pull/12593/files after pushing your latest version.

File: src/core_write.cpp (hunk 1 above)
Comment (practicalswift) [reply src/core_write.cpp]: I would revert this change in order to maximize the chance of merge.

File: test/functional/wallet_dump.py (hunk 10 above)
Comment (rex4539) [reply test/functional/wallet_dump.py]: Reverted. Not familiar with git rebase and such...

File: src/core_write.cpp (hunk 1 above)
Comment (rex4539) [reply src/core_write.cpp]: Reverted.

File: src/test/util_tests.cpp (hunk 8 above)
Comment (rex4539) [reply src/test/util_tests.cpp]: Reverted. Not familiar with git rebase and such...

File: src/test/util_tests.cpp (hunk 9 above)
Comment (rex4539) [reply src/test/util_tests.cpp]: Reverted. Not familiar with git rebase and such...

**Comment by practicalswift** [comment]
Please revert the unrelated change in `wallet_basic.py` (the Unicode check) and squash the commits into one commit.

File: test/functional/wallet_basic.py (hunk 4 above)
Comment (rex4539) [reply test/functional/wallet_basic.py]: This was very weird. It's some strange character encoding that my machine somehow automatically converts when the file is opened. I did several attempts to fix it using the command line and different text editors and I think I did it...

**Comment by rex4539** [comment]
Could you please post the exact command(s) I need to paste into the command line?

Not familiar at all with squashing :)

**Comment by practicalswift** [comment]
Seems like you accidentally re-introduced some of the previous reverts.

Please make sure that:
* There is only be one commit listed here: https://github.com/bitcoin/bitcoin/pull/12593/commits
* There are only wanted changes listed here: https://github.com/bitcoin/bitcoin/pull/12593/files

**Comment by rex4539** [comment]
ok, I think I reverted all the newly-accidental commits. All typos should be the correct ones now.

If you could just post the exact command(s) I need to paste into the command line so I could squash?

**Comment by rex4539** [comment]
@practicalswift Squashed and ready for merge :)

**Comment by practicalswift** [comment]
utACK 5fe43d6e2cc4b9bc35390cd4fe9f3d318440f732

File: src/leveldb/db/version_set.cc [hunk 13]
```diff
@@ -345,7 +345,7 @@ Status Version::Get(const ReadOptions& options,
 
   // We can search level-by-level since entries never hop across
   // levels.  Therefore we are guaranteed that if we find data
-  // in an smaller level, later levels are irrelevant.
+  // in a smaller level, later levels are irrelevant.
```
Comment (MarcoFalke) [inline src/leveldb/db/version_set.cc]: Please submit those upstream

**Comment by rex4539** [comment]
Unfortunately I messed up my repo while attempting to squash so I made a new PR #12716 with just the fixes.
